### PR TITLE
Don't conflict with upstream +test_name execution rules

### PR DESCRIPTION
**Pull request by chasinglogic**
This keeps the Ninja module working after https://jira.mongodb.org/browse/SERVER-44947 is completed

### CODE REVIEW

**Comment by chasinglogic** [comment]
https://mongodbcr.appspot.com/532970002/

This is the sister review where we add support for the `+test_name` and `+test_source_name` targets to SCons, the new generator, and to this generator.

**Review by markbenvenuto — APPROVED** [review]

File: build.py [hunk 1]
```diff
@@ -173,6 +173,10 @@ def __init__(self, name, env):
             if env.TargetOSIs('darwin'):
                 self.add_icecream_check()
             self.set_up_icecc()
+        if env.get('_NINJA_NO_TEST_EXECUTION'):
```
Comment (RedBeard0531) [inline build.py]: nit: `self.setup_test_execution = '_NINJA_NO_TEST_EXECUTION' not in env` or `not env.get('_NINJA_NO_TEST_EXECUTION')` if you expect it to be present but set to something Falsey.

This feels a lot like my pet peeve of
```py
if cond:
  return True
else:
  return False
```

Comment (chasinglogic) [reply build.py]: Done.

**Comment by chasinglogic** [comment]
We're planning on merging the change that will break the module without this today.

Ping: @RedBeard0531 @markbenvenuto

**Comment by markbenvenuto** [comment]
As I told you in person, you changes after your most recent patch don't work. You need to post a new commit before it can be merged.

**Comment by markbenvenuto** [comment]
I have verified it works on my mac. I will merge now.

**Pull request merged**